wordpress-mobile/WordPress-iOS
# Add emoji site icon creator

This PR adds an option to create a site icon using an emoji with a background color.

![emoji-picker-3](https://user-images.githubusercontent.com/4780/121676924-22630600-caad-11eb-92b7-be03665bba00.gif)

|   |   |
|---|---|
| ![Simulator Screen Shot - iPhone 12 Pro - 2021-06-11 at 12 03 52](https://user-images.githubusercontent.com/4780/121676914-2000ac00-caad-11eb-90fb-d76c7ebea260.png) | ![Simulator Screen Shot - iPhone SE (2nd generation) - 2021-06-11 at 11 50 51](https://user-images.githubusercontent.com/4780/121676931-255df680-caad-11eb-8c78-361669bca22e.png) |

I've updated the 'change site icon' alert sheet to now show an option for "Create from emoji". I've also limited the feature to iOS 14, as there are a few SwiftUI 2 features that make this work or perform better (in particular, `ScrollViewReader` and `LazyHStack`).

**To test**

* Build and run
* Tap the site icon for a site, and check you see the 'Create from Emoji' option in the sheet
* Try building an icon! Choose an emoji (check the emoji group picker bar also works to jump to a particular section of emoji) and a color, then tap Save.
* Try dismissing the screen without tapping Save and ensure nothing gets updated.
* Check the previous 'choose from my device' option for setting a site icon still works.
* Disable the feature flag and check it still works as before (the alert menu is slightly different before the feature flag).
* Test on an iOS 13 simulator and ensure the new option isn't shown.

## Regression Notes

1. Potential unintended areas of impact

* Breaking the site icon update flow, or the site icon Quick Start tour

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I manually tested both of these areas and they still worked as expected

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:
